hudi-common/src/main/java/org/apache/hudi/common/util/BaseFileUtils.java

```diff
@@ -67,6 +70,50 @@ public static BaseFileUtils getInstance(HoodieFileFormat fileFormat) {
     throw new UnsupportedOperationException(fileFormat.name() + " format not supported yet.");
   }
 
+  /**
+   * Aggregate column range statistics across files in a partition.
+   *
+   * @param fileRanges List of column range statistics for each file in a partition
+   */
+  public static <T extends Comparable<T>> HoodieColumnRangeMetadata<T> getColumnRangeInPartition(@Nonnull List<HoodieColumnRangeMetadata<T>> fileRanges) {
+    if (fileRanges.size() == 1) {
+      // Only one parquet file, we can just return that range.
+      return fileRanges.get(0);
+    }
+    // There are multiple files. Compute min(file_mins) and max(file_maxs)
+    return fileRanges.stream()
+        .sequential()
+        .reduce(BaseFileUtils::mergeRanges).get();
```

Comment:
we should be merging across files on a given column

Reply:
yea, apologies, current logic is incorrect. i'm going to fix it.

Reply:
fixed